library/std/tests/path.rs

```diff
@@ -1976,3 +1983,102 @@ fn clone_to_uninit() {
     unsafe { a.clone_to_uninit(ptr::from_mut::<Path>(&mut b).cast()) };
     assert_eq!(a, &*b);
 }
+
+// Test: Only separators (e.g., "/" or "\\")
+// This test checks how Path handles a string that consists only of path separators.
+// It should recognize the root and not treat it as a normal component.
+#[test]
+fn test_only_separators() {
+    let path = Path::new("/////");
+    assert!(path.has_root());
+    assert_eq!(path.iter().count(), 1);
+    assert_eq!(path.parent(), None);
+}
+
+// Test: Non-ASCII/Unicode
+// This test verifies that Path can handle Unicode and non-ASCII characters in the path.
+// It ensures that such paths are not rejected or misinterpreted.
+#[test]
+fn test_non_ascii_unicode() {
+    let path = Path::new("/tmp/❤/🚀/file.txt");
+    assert!(path.to_str().is_some());
+    assert_eq!(path.file_name(), Some(OsStr::new("file.txt")));
+}
+
+// Test: Embedded null bytes
+// This test checks that Path can be constructed from a byte slice containing a null byte (on Unix).
+// It ensures that null bytes are not treated as string terminators.
+#[test]
+fn test_embedded_null_byte() {
+    use std::ffi::OsStr;
+    let bytes = b"foo\0bar";
+    let os_str = OsStr::from_bytes(bytes);
+    let path = Path::new(os_str);
+    assert!(path.as_os_str().as_bytes().contains(&0));
```

Comment:
This relies on [OsStrExt](https://doc.rust-lang.org/std/os/unix/ffi/trait.OsStrExt.html#tymethod.as_bytes)

Reply:
It might be a little much so I just removed it for now. 🙇